### PR TITLE
Delete response

### PR DESCRIPTION
**Pull request by make-it-git**
## Overview

| Q                 | A
| Bugfix?           | no
| New feature?      | yes
| Breaking Changes? | yes
| Fixed issues      | #65 

## Notes

Need to find some API which does not allow deletion to verify what is returned

...

Notify the following people: @darrynten

### CODE REVIEW

File: src/Request/RequestHandler.php [hunk 1]
```diff
@@ -157,6 +157,12 @@ public function handleRequest(string $method, string $uri, array $options, array
             $this->handleException($exception);
         }
 
+        // For DELETE we should check response's HTTP code
+        // So we return response itself
+        if ($method === 'DELETE') {
+            return $response;
+        }
```
Comment (darrynten) [inline src/Request/RequestHandler.php:164]: @make-it-git what does the response from delete look like?

**Comment by darrynten** [comment]
Hi @make-it-git there's a conflict on this PR :) thanks

File: src/Request/RequestHandler.php (hunk 1 above)
Comment (make-it-git) [reply src/Request/RequestHandler.php:164]: Empty response with http code 204

File: src/BaseModel.php [hunk 2]
```diff
@@ -203,16 +203,24 @@ public function get(string $id)
...
+        // On success it returns 204 HTTP code with empty body
+        $response = $this->request->request(
+            $this->featureMethods['delete'],
+            $this->endpoint,
+            sprintf('Delete/%s', $id)
+        );
+        if ($response->getStatusCode() === 204) {
+            return true;
+        }
+        return false;
```
Comment (darrynten) [inline src/BaseModel.php:223]: This line is not being reached by unit tests @make-it-git

**Review by darrynten — CHANGES_REQUESTED** [review]

File: src/BaseModel.php (hunk 2 above)
Comment (make-it-git) [reply src/BaseModel.php:223]: Done, do not no yet how to test delete() correctly so for now tests for http code 300 (which is invalid, of course)

**Comment by darrynten** [comment]
Test is failing at

CompanyNoteModelTest::testDelete
true does not match expected type "object".

**Comment by make-it-git** [comment]
@darrynten merge is available

**Review by darrynten — APPROVED** [review]
Looking good! @fergusdixon @igorsergiichuk @kiproping this is what we've been waiting for 😀

**Comment by kiproping** [comment]
Yaaaaaay

**Pull request merged**